week2/homework/src/index.js

```diff
@@ -1,3 +1,112 @@
 'use strict';
-
 // TODO: Write the homework code in this file
+'use strict';
+const { readFile, writeFile, appendFile } = require('fs');
+const { promisify } = require('util');
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+const appendFileWithPromise = promisify(appendFile);
+
+const TODO_FILE = 'todo.json';
+
+async function main() {
+  const [, , cmd, ...args] = process.argv;
+  switch (cmd) {
+    case 'list': {
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
+      console.info(todos);
+      break;
+    }
+
+    case 'add': {
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
+      const newTodo = args.join(' ');
+      todos.push(newTodo);
+      await writeFileWithPromise(TODO_FILE, JSON.stringify(todos));
+      break;
+    }
+
+    case 'remove': {
+      const [, , cmd, baseIndex, ...args] = process.argv;
+      const index = parseInt(baseIndex);
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
```

Comment:
I see those 2 lines repeated several times: 

```js
const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
const todos = JSON.parse(data);
```
Maybe that could go in a function?